File: typescript/base-hmi-types/src/screens/widgets/HmiWidgetBase.ts
import { HmiColor, HmiFont, HmiHorizontalAlignment, HmiLineStyle, HmiProperty, HmiVerticalAlignment } from "../base.js";
import { HmiFillPattern } from "../base/HmiFillPattern.js";
import { HmiGradientDirection } from "../base/HmiGradientDirection.js";
import { HmiSimpleScreenItemBase } from "../base/HmiSimpleScreenItemBase.js";

export abstract class HmiWidgetBase extends HmiSimpleScreenItemBase {
  font?: HmiFont;
  horizontalAlignment?: HmiProperty<HmiHorizontalAlignment>;
  verticalAlignment?: HmiProperty<HmiVerticalAlignment>;
  focusColor?: HmiProperty<HmiColor>;
  focusWidth?: HmiProperty<number>;
  backFillPattern?: HmiProperty<number>;
  fillPattern?: HmiProperty<HmiFillPattern>;
  cornerRadius?: HmiProperty<number>;
  cornerStyle?: HmiProperty<number>;
  edgeStyle?: HmiProperty<HmiLineStyle>;
  firstGradientColor?: HmiProperty<HmiColor>;
  firstGradientOffset?: HmiProperty<number>;
  middleGradientColor?: HmiProperty<HmiColor>;
  secondGradientColor?: HmiProperty<HmiColor>;
  secondGradientOffset?: HmiProperty<number>;
  useFirstGradient?: HmiProperty<boolean>;
  useSecondGradient?: HmiProperty<boolean>;
  gradientDirection?: HmiProperty<HmiGradientDirection>;
  useDesignColorSchema?: HmiProperty<boolean>;
  useDesignShadowSettings?: HmiProperty<boolean>;
}

File: src/BaseHmiTypes/Screens/Widgets/HmiWidgetBase.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Widgets;

public abstract class HmiWidgetBase : HmiSimpleScreenItemBase
{
    public HmiFont? Font { get; set; }

    public HmiProperty<HmiHorizontalAlignment>? HorizontalAlignment { get; set; }

    public HmiProperty<HmiVerticalAlignment>? VerticalAlignment { get; set; }

    public HmiProperty<int>? TextWrapping { get; set; }

    public HmiProperty<int>? TextTrimming { get; set; }

    public HmiProperty<HmiColor>? FocusColor { get; set; }

    public HmiProperty<double>? FocusWidth { get; set; }

    public HmiProperty<int>? BackFillPattern { get; set; }

    public HmiProperty<HmiFillPattern>? FillPattern { get; set; }

    public HmiProperty<int>? CornerRadius { get; set; }

    public HmiProperty<int>? CornerStyle { get; set; }

    public HmiProperty<HmiLineStyle>? EdgeStyle { get; set; }

    public HmiProperty<HmiColor>? FirstGradientColor { get; set; }

    public HmiProperty<double>? FirstGradientOffset { get; set; }

    public HmiProperty<HmiColor>? MiddleGradientColor { get; set; }

    public HmiProperty<HmiColor>? SecondGradientColor { get; set; }

    public HmiProperty<double>? SecondGradientOffset { get; set; }

    public HmiProperty<bool>? UseFirstGradient { get; set; }

    public HmiProperty<bool>? UseSecondGradient { get; set; }

    public HmiProperty<HmiGradientDirection>? GradientDirection { get; set; }

    public HmiProperty<bool>? UseDesignColorSchema { get; set; }

    public HmiProperty<bool>? UseDesignShadowSettings { get; set; }
}
